src/tools/rustfmt/src/parse/session.rs

```diff
@@ -231,7 +199,21 @@ impl ParseSess {
     }
 
     pub(crate) fn set_silent_emitter(&mut self) {
-        self.parse_sess.dcx = DiagCtxt::with_emitter(silent_emitter());
+        // Ideally this invocation wouldn't be necessary and the fallback bundle in
+        // `self.parse_sess.dcx` could be used.
```

Comment:
Just to make sure I understand things correctly, is the reason this can't be used because of the lock that you mentioned in [compiler/rustc_errors/src/emitter.rs](https://github.com/rust-lang/rust/pull/121301/files#diff-a35029c589a6b5b002ecf598efb24d174232e719ecbc6f741a58eee8509cd6a5)? If so, could you also capture that context here.

Reply:
Added a note to this effect.